pkg/flare/strip.go

```diff
@@ -27,43 +27,35 @@ var blankRegex = regexp.MustCompile(`^\s*$`)
 var replacers []replacer
 
 func init() {
-	apiKeyReplacer = replacer{
-		regex: regexp.MustCompile(`(?m)^\s*api_key:( *['"]?\w+['"]? ?,?)+$`),
-		replFunc: func(b []byte) []byte {
-			s := string(b)
-			s = strings.TrimSuffix(s, `"`)
-			s = strings.TrimSuffix(s, `'`)
-			replacement := "api_key: **************************" + s[len(s)-5:]
-			return []byte(replacement)
-		},
-	}
-	dockerAPIKeyReplacer = replacer{
-		regex: regexp.MustCompile(`DD_API_KEY=['"]?\w+['"]?`),
-		replFunc: func(b []byte) []byte {
-			s := string(b)
-			s = strings.TrimSuffix(s, `"`)
-			s = strings.TrimSuffix(s, `'`)
-			replacement := "DD_API_KEY=**************************" + s[len(s)-5:]
-			return []byte(replacement)
-		},
+	apiKeyReplacer := replacer{
+		regex: regexp.MustCompile(`[a-fA-F0-9]{27}([a-fA-F0-9]{5})`),
+		repl:  []byte(`***************************$1`),
 	}
 	uriPasswordReplacer = replacer{
-		regex: regexp.MustCompile(`(.*\ ["']?[A-Za-z0-9]+)\:\/\/([A-Za-z0-9_]+)\:(.+)\@`),
-		repl:  []byte(`$1://$2:********@`),
+		regex: regexp.MustCompile(`\:\/\/([A-Za-z0-9_]+)\:(.+)\@`),
+		repl:  []byte(`://$1:********@`),
 	}
 	passwordReplacer = replacer{
-		regex: regexp.MustCompile(`( *(\w|_)*pass(word)?:).+`),
+		regex: matchYAMLKeyPart(`pass(word)?`),
 		repl:  []byte(`$1 ********`),
 	}
 	tokenReplacer = replacer{
-		regex: regexp.MustCompile(`( *(\w|_)*token:).+`),
+		regex: matchYAMLKeyPart(`token`),
```

Comment:
Will that replace the whole line ?

Reply:
It should replace everything after a YAML key containing `token` in it's name